src/runtime/entitlements-manager.js

```diff
@@ -237,33 +237,41 @@ export class EntitlementsManager {
 
   /**
    * Sends a pingback that marks a metering entitlement as used.
-   * @param {!Entitlements} entitlements
+   * @param {!Entitlement|null} entitlement
+   * @param {JsonObject|null|undefined} jwtContents
    */
-  consumeMeter_(entitlements) {
-    const entitlement = entitlements.getEntitlementForThis();
+  consumeMeter_(entitlement, jwtContents) {
```

Comment:
I don't think you need to pass in this `jwtContents` since it's the `entitlement.subscriptionToken`. Could just do the JWT decoding as part of this function

Reply:
Nvm, I see you also use it below. Maybe we can move the decode function to be inside the `Entitlements` object

Reply:
agree is there a case where we need to pass in a jwt that is not from the subscriptionToken? - if there isn't then we should get it internally to this function.

Reply:
JWT decoding looks like [it's synchronous](https://github.com/subscriptions-project/swg-js/blob/f9ff7b33d81be0804d050556c7c8038a848d0588/src/utils/jwt.js#L39). What I would propose is passing the decoded token into the `Entitlements` constructor (or let the Entitlements do the decoding), this way you can just expose a `getToken` function to access it.

This way, you're only decoding once when you create the entitlement and you don't have to pass around this additional `jwtContents` because you can just access it off the `entitlement` variable you already have access to.

How does that sound?

Reply:
It sounds quite awesome, actually. Done.

Reply:
My "Done" submitted prematurely. The commit with this refactor is now in for your review.